src/libecalc/core/models/compressor/results.py

```diff
@@ -373,14 +359,120 @@ def from_result_list_to_dto(
+            outlet_stream_condition_train.actual_rate_m3_per_hr = [
+                result_list[t].stage_results[-1].mass_rate_kg_per_hour / result_list[t].outlet_stream.density_kg_per_m3
+                for t in range(len(result_list))
+                if result_list[t].stage_results[-1].mass_rate_kg_per_hour is not None
+                and result_list[t].outlet_stream is not None
+            ]
+            outlet_stream_condition_train.actual_rate_before_asv_m3_per_hr = [
+                result_list[t].stage_results[-1].mass_rate_kg_per_hour / result_list[t].outlet_stream.density_kg_per_m3
+                for t in range(len(result_list))
+                if result_list[t].stage_results[-1].mass_rate_kg_per_hour is not None
+                and result_list[t].outlet_stream is not None
+            ]  #
+            outlet_stream_condition_train.standard_rate_sm3_per_day = [
+                compressor_stage_result[0].outlet_stream_condition.standard_rate_before_asv_sm3_per_day[t]
+                for t in range(len(result_list))
+            ]
+            outlet_stream_condition_train.standard_rate_before_asv_sm3_per_day = (
+                outlet_stream_condition_train.standard_rate_sm3_per_day
+            )
+            outlet_stream_condition_train.density_kg_per_m3 = [
+                result_list[t].outlet_stream.density_kg_per_m3
+                if result_list[t].outlet_stream is not None
+                and result_list[t].outlet_stream.density_kg_per_m3 is not None
+                else np.nan
+                for t in range(len(result_list))
+            ]
+            outlet_stream_condition_train.kappa = [
+                result_list[t].outlet_stream.kappa
+                if result_list[t].outlet_stream is not None and result_list[t].outlet_stream.kappa is not None
+                else np.nan
+                for t in range(len(result_list))
+            ]
+            outlet_stream_condition_train.z = [
+                result_list[t].outlet_stream.z
+                if result_list[t].outlet_stream is not None and result_list[t].outlet_stream.z is not None
+                else np.nan
+                for t in range(len(result_list))
+            ]
+            outlet_stream_condition_train.temperature_kelvin = [
+                result_list[t].outlet_stream.temperature_kelvin
+                if result_list[t].outlet_stream is not None
+                and result_list[t].outlet_stream.temperature_kelvin is not None
+                else np.nan
+                for t in range(len(result_list))
+            ]
+
+        return inlet_stream_condition_train, outlet_stream_condition_train, compressor_stage_result
 
     model_config = ConfigDict(extra="forbid")
 
     @property
     def failure_status(self):
         if not all(r.is_valid for r in self.stage_results):
-            for i, stage in enumerate(self.stage_results):
+            for stage in self.stage_results:
                 if not stage.is_valid:
```

Comment:
After target pressure status is moved to train - stage.within_capacity and stage.is_valid is the same, so fix this if-statement...